core/src/main/scala-3/toml/derivation/DerivedProductCodec.scala

```diff
@@ -0,0 +1,62 @@
+package toml
+package derivation
+import toml.Codec.Defaults
+import toml.Codec.Index
+import shapeless3.deriving.*
+
+trait DerivedProductCodec[P] extends Codec[P]
+object DerivedProductCodec:
+  given codecGen[P](
+    using inst: K0.ProductInstances[Codec, P],labelling: Labelling[P], d:DefaultParams[P]
+  ): DerivedProductCodec[P]  with
+    def apply(value: Value, defaults: Defaults, index: Int): Either[Parse.Error, P] =
+      type Acc = (Value, (/*labels: */Seq[String], Index), Parse.Error)
+      inst.unfold[Acc]((value, (labelling.elemLabels, index), (Nil, "")))(
```

Comment:
It uses `unfold` instead of `constructA` because Scala 2 derivation performs stateful operation on labels and index.